src/mlang/mpp_ir/mpp_ir_to_bir.ml

```diff
@@ -470,31 +470,30 @@ let create_combined_program (m_program : Mir_interface.full_program)
     Bir.program =
   try
     let mpp_program = List.rev mpp_program in
-    let decl_to_extract =
-      try
-        List.find
-          (fun decl -> decl.Mpp_ir.name = mpp_function_to_extract)
-          mpp_program
-      with Not_found ->
-        Errors.raise_error
-          (Format.asprintf "M++ function %s not found in M++ file!"
-             mpp_function_to_extract)
-    in
-    let ctx, statements =
-      translate_mpp_function mpp_program m_program decl_to_extract []
-        emtpy_translation_ctx
+    let ctx, mpp_functions =
+      List.fold_left
+        (fun (ctx, function_map) mpp_func ->
+          let ctx, statements =
+            translate_mpp_function mpp_program m_program mpp_func [] ctx
+          in
+          (ctx, Bir.FunctionMap.add mpp_func.name statements function_map))
+        (emtpy_translation_ctx, Bir.FunctionMap.empty)
+        mpp_program
     in
+    ignore
+      (try Bir.FunctionMap.find mpp_function_to_extract mpp_functions
```

Comment:
`Bir.FunctionMap.mem` for a more natural unit if-then.

Reply:
Of course! thanks!